### ISSUE TITLE
Publisher tag in oai_dc format

### ISSUE THREAD

**Issue by tcoupin** [IQSS/dataverse#8739]
When oai_dc format is used for harvesting, publisher tag is used as producerName: https://github.com/IQSS/dataverse/blob/v5.10.1/src/main/resources/db/migration/afterMigrate__1-7256-upsert-referenceData.sql#L34

But the citation block in the datasets list does not show it because it uses the distributor name field:
- https://github.com/tcoupin/dataverse/blob/8927ff09c48c59f5f78f0c4b71adb3d6f01f9015/src/main/java/edu/harvard/iq/dataverse/DataCitation.java#L211
- https://github.com/tcoupin/dataverse/blob/8927ff09c48c59f5f78f0c4b71adb3d6f01f9015/src/main/java/edu/harvard/iq/dataverse/DataCitation.java#L141
- https://github.com/tcoupin/dataverse/blob/8927ff09c48c59f5f78f0c4b71adb3d6f01f9015/src/main/java/edu/harvard/iq/dataverse/DataCitation.java#L758

OAI-PMH response:

![image](https://user-images.githubusercontent.com/7457386/170275090-5c92c222-695a-49bf-a237-10fc2f4e3e2d.png)

Citation:
![image](https://user-images.githubusercontent.com/7457386/170275183-5c33c6e2-1126-497a-a1b9-028f299ef73a.png)

Citation from remote dataverse:
![image](https://user-images.githubusercontent.com/7457386/170276057-593fd8cd-6629-4d9d-9d9f-1a9faecab203.png)

solr:
![image](https://user-images.githubusercontent.com/7457386/170275790-79a0cb34-fa00-4497-a21a-31bc78e5ca72.png)

**What steps does it take to reproduce the issue?**

* When does this issue occur?

* Which page(s) does it occurs on?

* What happens?

* To whom does it occur (all users, curators, superusers)?

* What did you expect to happen?

**Which version of Dataverse are you using?**

5.10.1

**Any related open or closed issues to this bug report?**

**Screenshots:**

No matter the issue, screenshots are always welcome.

To add a screenshot, please use one of the following formats and/or methods described here:

* https://help.github.com/en/articles/file-attachments-on-issues-and-pull-requests
*

**Comment by tcoupin** [comment]
Suggestion: update table foreignmetadatafieldmapping to map publisher tag to distributorname

**Comment by pdurbin** [comment]
@tcoupin to you want to go ahead and make a pull request for this? (Thanks for all the pull requests so far!) It would be a way to get the discussion going, perhaps.

Also, I'm thinking this would be a great example of where we could use some automated testing, to make some assertions of what happens with harvesting if you put a value in this or that field.

**Comment by tcoupin** [comment]
Yes I can do a PR. We already have made this SQL update on dataverse.ird.fr

**Comment by mreekie** [comment]
Grooming:
- This looks to fix a harvesting issue.
- Associating the deliverable.
